hooks/pre/12-clear_pulp_content.rb

```diff
@@ -0,0 +1,74 @@
+# See bottom of the script for the command that kicks off the script
+require 'English'
+
+def reset
+  stop_services
+  reset_pulp
+end
+
+def empty_mongo
+  logger.info "Dropping Pulp database!"
+
+  mongo_config = load_mongo_config
+  if remote_host?(mongo_config[:host])
+    empty_remote_mongo(mongo_config)
+  else
+    execute(
+      [
+        'systemctl start rh-mongodb34-mongod',
+        "mongo #{mongo_config[:database]} --eval 'db.dropDatabase();'",
+      ]
+    )
+  end
+end
+
+def load_mongo_config
+  config = {}
+  seeds = param_value('katello', 'pulp_db_seeds')
+  seed = seeds.split(',').first
+  host, port = seed.split(':') if seed
+  config[:host] = host || 'localhost'
+  config[:port] = port || '27017'
+  config[:database] = param_value('katello', 'pulp_db_name') || 'pulp_database'
+  config[:username] = param_value('katello', 'pulp_db_username')
+  config[:password] = param_value('katello', 'pulp_db_password')
+  config[:ssl] = param_value('katello', 'pulp_db_ssl') || false
+  config[:ca_path] = param_value('katello', 'pulp_db_ca_path')
+  config[:ssl_certfile] = param_value('katello', 'pulp_db_ssl_certfile')
+  config
+end
```

Comment:
Most functions here (and similarly in `hooks/pre/10-reset_data.rb` and in `hooks/pre/13-clear_pulpcore_content.rb`) could be moved to `hooks/boot/01-kafo-hook-extensions.rb` or something new like `hooks/boot/05-data-reset-helpers.rb` to prevent duplication of code.

Reply:
Before you merged that, it was actually going to be my comment that we should just drop Pulp 2 before migrating things over.

Reply:
Moving to a dedicated data helpers file in `boot` makes sense.

Reply:
@wbclark Do you want to tackle the re-factor mentioned here in this PR or a follow up?